addon/model.js

```diff
@@ -226,10 +221,14 @@ export default class MegamorphicModel extends EmberObject {
       // TODO: disconnect recordData -> childRecordData in the case of nested model -> primitive
       // anything -> undefined | primitive
       delete this._cache[key];
-      super.notifyPropertyChange(key);
+      this._deferProprtyChange(key);
     }
   }
 
+  _deferProprtyChange(key) {
+    deferPropertyChange(this._store, this, key);
```

Comment:
It seems like defer logic is only called here, I'm curious about the tradeoffs of having the defer logic in the util method vs putting it on the store. It seems that we create a weak map to map stores to changes, but to me it seems natural for the store to track/manage this?

Reply:
as we discussed OOB; `WeakMap` is used for private state only; conceptually it is managed by the store